### PR TITLE
Bugfix: GitHub MPC Configuration #419

### LINKED ISSUE #419: Bug: GitHub integration setup does not confirm auth or repo access
## Summary
`opensre integrations setup github` saves the GitHub MCP configuration but does not tell the user whether authentication actually succeeded or what repositories the configured token can access.

In the observed flow, the command prompts for transport, MCP URL, PAT/auth token, and toolsets, then only prints a saved-config message. That leaves the user unsure whether the token works, whether upstream auth was used, and whether later repo-scoped tools will have the necessary access.

## Reproduction
1. Run `opensre integrations setup`
2. Choose `github`
3. Select `Streamable HTTP`
4. Use `https://api.githubcopilot.com/mcp/`
5. Enter a GitHub PAT / auth token or leave it blank if the server authenticates upstream
6. Accept the default toolsets
7. Observe that the command ends with a success message for saving config only

## Actual behavior
The flow writes the config and prints a saved path, but it does not validate the connection or show repository access scope.

## Expected behavior
After GitHub configuration, the CLI should:
- clearly report whether the configuration was validated successfully
- show which GitHub identity was authenticated
- show what repository access is available to the integration, such as repo count, org/repo scope, and/or a representative list of accessible repos
- surface auth or permission failures before treating setup as successful

## Notes
- The legacy setup path in `app/integrations/cli.py` currently saves GitHub config directly.
- The newer wizard flow in `app/cli/wizard/flow.py` already calls `validate_github_mcp_integration(...)` before saving, so this looks like a parity gap as well as a missing UX detail.
- Current validation in `app/integrations/github_mcp.py` verifies tool discovery plus `get_me`, but it does not appear to report repository scope/access.

## Acceptance criteria
- `opensre integrations setup github` validates the GitHub MCP config before printing success.
- Success output includes authenticated identity and repo-access feedback.
- Failures explain whether the problem is connectivity, auth, or insufficient scope.
- The CLI does not leave the user guessing whether the integration is usable for repo operations.
- Tests cover both validated-success and validation-failure UX.

### PR DESCRIPTION
**Pull request by Yashkapure06**
Fixes #419 

 

#### Describe the changes you have made in this PR -

- Added GitHub MCP config validation before save for opensre integrations setup github (fails fast; no "Saved" on failure).
- Improved validation to report authenticated identity and repo-access readiness with clear failure categories (connectivity/auth/toolset/repo access).
- Added a clean Rich report for setup/wizard with brief/standard/expanded views.
- Added repo-access verification options (auto / your repos / starred / search user) and an optional public/private filter (best-effort based on payload fields).
- Updated defaults/toolsets and added tests covering success/failure UX and validation behavior.

### Screenshots of the UI changes (If any) -

image

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

This PR fixes confusing GitHub MCP onboarding where setup could appear successful without confirming repo operations. I implemented a shared validator (validate_github_mcp_config) that checks MCP connectivity/tool availability, authenticates via get_me, then probes repo access using the best available tool (listing/search/profile fallback). The CLI and wizard now run validation before saving and display a clean summary of identity + repo-access readiness, with optional expanded details. I chose this approach to keep behavior consistent across entrypoints, keep validation fast (single probe), and ensure failures are categorized and actionable. Tests cover both success and failure UX.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

### CODE REVIEW

File: app/integrations/cli.py [hunk 1]
```diff
@@ -71,21 +77,46 @@ def _p(label: str, default: str = "", secret: bool = False) -> str:
     return result.strip() or default
 
 
-def _die(msg: str) -> NoReturn:
+def _die(msg: str) -> None:
```
Comment (Devesh36) [inline app/integrations/cli.py]: _die() is typed incorrectly

It still always exits the program via sys.exit(1), so it should be annotated as NoReturn, not None.
Changing it to None breaks static analysis and forces dead return statements after _die().
Fix: change the return type back to -> NoReturn.

Comment (Yashkapure06) [reply app/integrations/cli.py]: @Devesh36 thank you for pointing out!

**Comment by Yashkapure06** [comment]
@Devesh36 can you please help me to solve this

`FAILED tests/cli_smoke_test.py::test_onboard_interactive_smoke - AssertionError: Timed out waiting for CLI exit.`

**Comment by Devesh36** [comment]
> @Devesh36 can you please help me to solve this
> 
> `FAILED tests/cli_smoke_test.py::test_onboard_interactive_smoke - AssertionError: Timed out waiting for CLI exit.`

ya sure let me check it

**Comment by davincios** [comment]
Addressed the remaining review feedback on the latest head `e0208b26`.

- restored legacy `opensre integrations setup vercel` support (`_setup_vercel` + handler)
- fixed the GitHub MCP repo-probe failure detail so it reports only the tools actually attempted for the selected view
- merged current `main` and resolved the onboarding/system-metrics conflicts
- verified the previously failing `tests/cli_smoke_test.py::test_onboard_interactive_smoke` now passes

Local verification on this head:
- focused regression tests: `20 passed`
- `make lint`
- `make typecheck`
- `make test-cov` (`2503 passed, 1 skipped`)

CI is rerunning on the pushed branch now.
